### PR TITLE
lecture 1 solutions done

### PR DESCRIPTION
**Pull request by gantu**
### Solutions for Lecture {NUMBER}

cc @chshersh @vrom911

### CODE REVIEW

File: src/Lecture1.hs [hunk 1]
```diff
@@ -63,7 +65,8 @@ sumOfSquares x y = error "TODO!"
 
 -}
 -- DON'T FORGET TO SPECIFY THE TYPE IN HERE
-lastDigit n = error "lastDigit: Not implemented!"
+lastDigit :: Integer -> Integer
+lastDigit n = mod (abs n) 10
```
Comment (chshersh) [inline src/Lecture1.hs:69]: Absolutely correct 👍🏻 
Alternatively, you can use the infix form of a function by putting it inside backticks :slightly_smiling_face: 
```suggestion
lastDigit n = abs n `mod` 10
```

File: src/Lecture1.hs [hunk 2]
```diff
@@ -104,7 +115,11 @@ and finds a sum of the numbers inside this string.
 
 The string contains only spaces and/or numbers.
 -}
-strSum str = error "TODO"
+strSum :: String -> Int
+strSum str = sum lw
+    where 
+        w = words str
+        lw = map (\x -> read x :: Int) w
```
Comment (chshersh) [inline src/Lecture1.hs:122]: As you specify the top-level type signature for this function, the compiler would be able to figure out what type `read` function has in this case due to type inference, so you can skip specifying it in the body:

```suggestion
        lw = map read w
```

File: src/Lecture1.hs [hunk 3]
```diff
@@ -119,4 +134,8 @@ and lower than 6 elements (4, 5, 6, 7, 8 and 9).
 
 🕯 HINT: Use recursion to implement this function.
 -}
-lowerAndGreater n list = error "TODO"
+lowerAndGreater :: Integer -> [Integer] -> String
+lowerAndGreater n list = show n ++ " is greater than " ++ show lw ++ " elements and lower than " ++ show gr ++ " elements"
+    where
+        gr = length $ filter ( > n) list
+        lw = length $ filter ( < n) list
```
Comment (chshersh) [inline src/Lecture1.hs:141]: That's correct! 💯 

However, with this solution, you will walk through the list twice.
With the recursive solution and manual check on every element in the list on both conditions at the same time (without `filter` function), you would be able to write it more performant with only one traversal of the list 🙂

**Review by chshersh — APPROVED** [review]
Congrats! The first homework is complete 👏🏼 
Hope you enjoyed the course so far. Keep up the great work, and feel free to ask any questions if you have 👌🏼

File: src/Lecture1.hs (hunk 3 above)
Comment (gantu) [reply src/Lecture1.hs:141]: you mean tail - recursion with two aggregates ?

Comment (chshersh) [reply src/Lecture1.hs:141]: @gantu Yes, you're correct 👌🏻 That's exactly what I mean 🙂

**Pull request merged**